sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -2852,6 +2927,112 @@ public void UnexpectedException(string errorMessage)
             }
         }
 
+        /// <summary>
+        ///    Indicates that an <see cref="EventProcessor{TPartition}" /> instance has dispatched a batch of events to the processing handler, writing into a stack allocated
+        ///   <see cref="EventSource.EventData"/> struct to avoid the parameter array allocation on the WriteEvent methods.
+        /// </summary>
+        ///
+        /// <param name="eventId">The identifier of the event.</param>
+        /// <param name="partitionId">The identifier of the Event Hub partition associated with the event batch.</param>
+        /// <param name="identifier">A unique name used to identify the event processor.</param>
+        /// <param name="eventHubName">The name of the Event Hub that the processor is associated with.</param>
+        /// <param name="consumerGroup">The name of the consumer group that the processor is associated with.</param>
+        /// <param name="operationId">An artificial identifier for the handler invocation.</param>
+        /// <param name="eventCount">The number of events in the batch that was passed to the processing handler.</param>
+        ///
+        [NonEvent]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe void EventProcessorProcessingHandlerStartCore(int eventId,
```

Comment:
Just to play devil's advocate, why not just define the core event with all string parameters and have it call directly into WriteEvent. I realize now that this same comment could be equally applied to the Service Bus events.

Reply:
Eh, I guess this could have a negative impact for someone wanting to consume the individual typed parameters rather than just look at logs. Carry on!

Reply:
Truthfully, I'm not sure.   That would seem to be a reasonable question.   I guess the best answer is that I just didn't think of it.